cmd/kube-controller-manager/app/core.go

```diff
@@ -178,6 +178,7 @@ func startEndpointController(ctx ControllerContext) (bool, error) {
 	go endpointcontroller.NewEndpointController(
 		ctx.InformerFactory.Core().V1().Pods(),
 		ctx.InformerFactory.Core().V1().Services(),
+		ctx.InformerFactory.Core().V1().Endpoints(),
```

Comment:
QPS drops substantially in my testing. I also don't see spurious updates I was seeing before (during a resync, about 20% of the endpoints would cause a PUT even though nothing changed - not sure why).